tests/ifsbench/ifsbench_run.py

```diff
@@ -0,0 +1,366 @@
+#! @IFSBENCH_PYTHON@
+
+# (C) Copyright 2024- ECMWF.
+#
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+from contextlib import nullcontext
+from pathlib import Path
+import re
+from tempfile import TemporaryDirectory
+from typing import List, Dict
+
+import click
+import yaml
+
+from ifsbench import (
+    Benchmark,
+    cli,
+    debug,
+    DataFileStats,
+    DefaultApplication,
+    DefaultArch,
+    EnvHandler,
+    error,
+    Job,
+    PydanticConfigMixin,
+    PydanticDataFrame,
+    ScienceSetup,
+    TechSetup
+)
+
+from ifsbench.data import FetchHandler, NamelistHandler, NamelistOverride, RenameHandler, RenameMode
+from ifsbench.validation import FrameCloseValidation
+
+class EclandResult(PydanticConfigMixin):
+    """
+    Ecland result class that can be serialised using the PydanticConfigMixin approach.
+    """
+
+    #: Numerical results of the run, stored as DataFrames (with corresponding
+    #: file name).
+    frames: Dict[str, PydanticDataFrame]
+
+    #: Standard out of the run.
+    stdout: str = None
+
+    #: Standard error of the run.
+    stderr: str = None
+
+    #: Walltime of the run in seconds.
+    walltime: float = None
+
+    @classmethod
+    def from_rundir(cls, run_dir: Path, **kwargs):
+        """
+        Build a result object from data in a run directory.
+
+        Params
+        ------
+        run_dir: pathlib.Path
+            The run directory.
+        kwargs:
+            If values are given for any of the EclandResult attributes, these
+            values are used instead of the extracted data from the run directory.
+        """
+
+        # Copy the kwargs dict so we can modify it.
+        kwargs = dict(kwargs)
+
+        if 'stderr' in kwargs:
+            # Extract the walltime from the stderr, if necessary.
+            match = re.search(r'wall-time\s*:\s*([0-9\.]+)s', kwargs['stderr'])
+            if match and 'walltime' not in kwargs:
+                kwargs['walltime'] = float(match.group(1))
+
+        # Gather the numerical data by parsing the netCDF files.
+        frames = {}
+
+        # ecLand writes output netCDF files of the form o_<some_name>.nc. Glob
+        # all of them and parse them.
+        paths = sorted(list(run_dir.glob('o_*.nc')))
+
+        for path in paths:
+            stats = DataFileStats(
```

Comment:
A short comment here on what the `DataFileStats` class provides would be good to understand what comes next.